### PR TITLE
Update top level docs

### PR DESCRIPTION
**Pull request by liustanley**
### What does this PR do?

- Removes Makefile since it's specific to Go repos
- Remove basic concepts file since it's a dupe of public docs
- Move common mistakes to /guides
- Add CONTRIBUTING.md and edit README.md

### Motivation

Open sourcing opentelemetry-examples

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,39 @@
...
-This repository includes resources for support, engineering and others to understand how Datadog support of OpenTelemetry works today.
-Available resources are:
-- [Basic concepts][1] for a basic introduction to OpenTelemetry and Datadog support of it.
-- [Common mistakes][2] for common mistakes or confusion end-users when using OpenTelemetry.
-- [Collector examples][3] for configuration examples of the OpenTelemetry Collector.
-
-[1]: ./basic-concepts.md
-[2]: ./common-mistakes.md
-[3]: collector-configurations
+This repository includes example applications and configurations for Datadog users, engineers, and support to understand how Datadog support of [OpenTelemetry][1] works today. These examples are intended to serve as references for learning how to integrate OpenTelemetry instrumented applications with Datadog products, and can be run independently to experiment with OpenTelemetry behaviors. Available resources are:
```
Comment (dineshg13) [inline README.md:3]: Convert to active voice. Suggestion:

```
The repository includes example applications and configurations for Datadog users, engineers, and support to understand how Datadog support of [OpenTelemetry](https://opentelemetry.io/) works today. These examples provide reference material for integrating OpenTelemetry instrumented applications with Datadog products and allow independent experimentation with OpenTelemetry behaviors. The available resources include:```

File: README.md [hunk 2]
```diff
@@ -1,11 +1,39 @@
...
+- [Configurations][15] for configuration examples of the Datadog Agent and the OpenTelemetry Collector.
+
+## Index
+
+| Example  | Description | Monitoring Capabilities |
+| ------------- | ------------- | ------------- |
+| [Game of Life][2]  | Go client-server app using gRPC, OTel Go and dd-trace-go instrumentations | Tracing gRPC and HTTP endpoints, OTLP metrics and logs, runtime metrics |
+| [Kafka Redis Messages][3] | Distributed app with Kafka and Redis, OTel and Datadog instrumentations | Kafka, Redis |
+| [Log Trace Correlation][4] | Go client-server app automatically generating traces and logs, OTel instrumentation with Datadog Agent | Log trace correlation |
+| [Manual Container Metrics][5] | Go server manually generating OTel container metrics, OTel Go instrumentation with OTel Collector | Container metrics correlating with traces in the trace app |
```
Comment (dineshg13) [inline README.md:14]: `trace app` may be clear for end user.

Comment (liustanley) [reply README.md:14]: Which line are you referring to?

Comment (dineshg13) [reply README.md:14]: `correlating with traces in the trace app`

**Review by dineshg13 — APPROVED** [review]

**Pull request merged**